src/transports/mod.rs

```diff
@@ -0,0 +1,129 @@
+//! This modules define the prototype of a transport and the dispatching between the different transports.
+//!
+//! This module use enum dispatch to avoid using trait objects and to save runtime costs.
+
+use std::{net::SocketAddr, time::Duration};
+
+use crate::{error::PeerNetError, network_manager::SharedPeerDB};
+
+use self::{quic::QuicTransport, tcp::TcpTransport};
+
+mod quic;
+mod tcp;
+
+pub use quic::QuicOutConnectionConfig;
+pub use tcp::TcpOutConnectionConfig;
+
+/// Define the different transports available
+/// TODO: Maybe try to fusion with the InternalTransportType enum above
+#[derive(Hash, Eq, PartialEq, Debug, Copy, Clone)]
+pub enum TransportType {
```

Comment:
I'm wondering if this should be extended with supporting for instance, multiple QUIC implementation

Reply:
It's possible right now, this enum is just used here to be used as an identifier for function that initialize a transport.